tests/Phinx/Migration/ManagerTest.php

```diff
@@ -5703,17 +5729,110 @@ public function testBreakpointsOperateAsExpected()
         while ($now == time()) {
         }
 
-        // reset all breakpoints
+        // Reset all breakpoints and toggle the most recent migration twice
         $this->manager->removeBreakpoints('production');
+        $this->manager->toggleBreakpoint('production', null);
+        $this->manager->toggleBreakpoint('production', null);
 
         // ensure breakpoint is not set
-        $versions = $this->manager->getEnvironment('production')->getVersionLog();
-        $this->assertEquals(0, end($versions)['breakpoint']);
+        $resetVersions = $this->manager->getEnvironment('production')->getVersionLog();
+        $this->assertEquals(0, reset($resetVersions)['breakpoint']);
+        $this->assertEquals(0, end($resetVersions)['breakpoint']);
+
+        // ensure no other data has changed.
+        foreach ($originalVersions as $originalVersionKey => $originalVersion) {
+            foreach ($originalVersion as $column => $value) {
+                if (!is_numeric($column)) {
+                    $this->assertEquals($value, $resetVersions[$originalVersionKey][$column]);
+                }
+            }
+        }
+
+        // Wait until the second has changed.
+        $now = time();
+        while ($now == time()) {
+        }
```

Comment:
Don't waste CPU. `usleep()` for 0.1s between checks or calculate remaining microseconds.

Maybe add helper method as you use this block multiple times.

Reply:
As the phinxlog only holds the start_time to 1s resolution, we need to wait until the next second so that we can demonstrate that changing the breakpoint in the phinxlog doesn't also change the start_time which has (for some reason) `ON UPDATE CURRENT_TIMESTAMP()`.

Reply:
But sleeping for 1s makes a lot more sense.

Reply:
It is better to 10 * (check time + wait 0.1s) than check time 123456 times using 100% CPU.

Reply:
Also, I you can use [time_sleep_until()](https://www.php.net/manual/en/function.time-sleep-until.php)

Reply:
The main issue is that the code has to wait 1s regardless of how we wait. The 1s delay is best as even though PHP has waited until the next second, any deviation between the clocks between PHP and mysql could mean that I am requesting changes to the phinxlog in the same 1s from mysql's perspective. By waiting 1s, we have waited 1s, so unless time runs differently in mysql, we will always be in the next second and so the test can adequately prove we are not altering the start_time because we are setting the breakpoint.